### PR TITLE
Update expected state calcs

### PR DESCRIPTION
**Pull request by ScaredIbis**
- add `calcPoolStatePreview` to `calculations` utils
- add `Pool.getPoolStatePreview` which uses `calcPoolStatePreview`
- update docs to include `Pool.getPoolStatePreview` and add deprecated tag to `Pool.getNextPoolState`

### CODE REVIEW

File: src/entities/pool.ts [hunk 1]
```diff
@@ -151,6 +166,30 @@ export default class Pool {
...
+					poolSwapLibraryByNetwork[this.chainId],
+					this.provider
+				)
+
+				const leverageAmountBytes = await contract.leverageAmount();
+				const convertedLeverage = await poolSwapLibrary.convertDecimalToUInt(leverageAmountBytes);
+
+				this.leverage = convertedLeverage.toNumber();
+			} catch (error) {
+				this.leverage = parseInt(name.split('-')?.[0] ?? 1);
```
Comment (dospore) [inline src/entities/pool.ts:189]: this way is king and admit it

File: src/types/types.ts [hunk 2]
```diff
@@ -32,3 +32,68 @@ export type StaticTokenInfo = {
...
+  ethers.BigNumber
+] & {
+  longMintSettlement: ethers.BigNumber;
+  longBurnPoolTokens: ethers.BigNumber;
+  shortMintSettlement: ethers.BigNumber;
+  shortBurnPoolTokens: ethers.BigNumber;
+  shortBurnLongMintPoolTokens: ethers.BigNumber;
+  longBurnShortMintPoolTokens: ethers.BigNumber;
+  updateIntervalId: ethers.BigNumber;
+}
```
Comment (dospore) [inline src/types/types.ts:52]: is this return from contracts?

File: src/utils/constants.ts [hunk 3]
```diff
@@ -16,3 +16,4 @@ export const TEST_QUOTE_TOKEN = {
     address: '0xa72276C7ecDc2D97b029F39Dd23fEf3Ea07D2ff8',
     decimals: TEST_TOKEN_DECIMALS,
 }
+
```
Comment (dospore) [inline src/utils/constants.ts:19]: !!!!

File: src/utils/helpers.ts [hunk 4]
```diff
@@ -18,3 +20,59 @@ export const encodeCommitParams: (
...
+      updateIntervalId: ethersBNtoBN(pendingCommits.updateIntervalId)
+    };
+  }
+
+/**
+ * attempt promise until it succeeds or the maximum number of allowed attempts is reached
+ *
+ * @returns a promise that will eventually error or resolve to the same type as the original promise
+ */
+export const attemptPromiseRecursively = async <T>({
```
Comment (dospore) [inline src/utils/helpers.ts:49]: the scary function is back and now in sdk

**Review by dospore — APPROVED** [review]
nothing I can see, if tests pass lgtm

File: src/types/types.ts (hunk 2 above)
Comment (ScaredIbis) [reply src/types/types.ts:52]: yeah I couldn't get the return type out of `PoolCommitter.totalPoolCommitments` and got sad to just did this :(

**Pull request merged**